scripts/openyurt-deployer/main_test.go

```diff
@@ -0,0 +1,275 @@
+		Master: "username@masterip",
+		Workers: Workers{
+			Cloud: []string{"username@cloudip"},
+			Edge:  []string{"username@edgeip"},
+		},
+	}
+
+	t.Logf("expected res: %s", expected)
+	// Compare the result with the expected value
+	if !jsonEqual(result, expected) {
+		t.Errorf("Expected %v, got %v", expected, result)
+	}
+}
+
+// jsonEqual checks if two JSON objects are equal.
+func jsonEqual(a, b interface{}) bool {
+	ajson, err := json.Marshal(a)
+	if err != nil {
+		return false
+	}
+
+	bjson, err := json.Marshal(b)
+	if err != nil {
+		return false
+	}
+
+	return string(ajson) == string(bjson)
+}
+
+func TestParseNodeInfo(t *testing.T) {
+
+	// Mock NodesInfo
+	mockNodesInfo := NodesInfo{
+		Master: "runner@127.0.0.1",
+		// Workers: Workers{
+		// 	Cloud: []string{"cloud@host"},
+		// 	Edge:  []string{"edge@host"},
+		// },
+	}
+
+	// Define a table of criteria
+	criteriaTable := map[string]string{
+		"Golang":     "1.19.10",
+		"containerd": "1.6.18",
+		"runc":       "1.1.4",
+		"CNI":        "1.2.0",
+	}
+
+	// Capture standard output
+	r, w, err := os.Pipe()
+	if err != nil {
+		t.Fatalf("Error creating pipe: %v", err)
+	}
+	defer r.Close()
+	defer w.Close()
+
+	// Redirect standard output to the pipe
+	oldStdout := os.Stdout
+	os.Stdout = w
+	defer func() {
+		os.Stdout = oldStdout
+	}()
+
+	// Call the function to be tested
+	nodeList := parseNodeInfo(mockNodesInfo)
+	t.Logf("nodeList: %v", nodeList)
+
+	// Capture stdout
+	w.Close()
+	var capturedOutput strings.Builder
+	_, err = io.Copy(&capturedOutput, r)
+	if err != nil {
+		t.Fatalf("Error reading from pipe: %v", err)
+	}
+	t.Logf("captop: %v", capturedOutput.String())
+
+	// Line split stdout
+	lines := strings.Split(capturedOutput.String(), "\n")
+	for _, line := range lines {
+		// Example: Check for keywords and versions
+		if strings.Contains(line, " Golang(version") {
+			t.Logf("line: %s", line)
+			if !strings.Contains(line, criteriaTable["Golang"]) {
+				t.Logf("failing: %s", line)
+				t.Errorf("Expected version %s not found in output for keyword %s", criteriaTable["Golang"], "Golang")
+			}
+		} else if strings.Contains(line, "containerd(version") {
+			t.Logf("line: %s", line)
+			if !strings.Contains(line, criteriaTable["containerd"]) {
+				t.Logf("failing: %s", line)
+				t.Errorf("Expected version %s not found in output for keyword %s", criteriaTable["containerd"], "containerd")
+			}
+		} else if strings.Contains(line, "runc(version") {
+			t.Logf("line: %s", line)
+			if !strings.Contains(line, criteriaTable["runc"]) {
+				t.Logf("failing: %s", line)
+				t.Errorf("Expected version %s not found in output for keyword %s", criteriaTable["runc"], "runc")
+			}
+		} else if strings.Contains(line, "CNI plugins(version") {
+			t.Logf("line: %s", line)
+			if !strings.Contains(line, criteriaTable["CNI"]) {
+				t.Logf("failing: %s", line)
+				t.Errorf("Expected version %s not found in output for keyword %s", criteriaTable["CNI"], "CNI")
+			}
+		}
+	}
+}
+
```

Comment:
there should not be any stranded commented blocks of code
either remove or restructure the code, or create an issue about why it is commented out, leaving a comment in the code (here)